migen/build/platforms/sinara/kasli_soc.py

```diff
@@ -0,0 +1,299 @@
+from migen.build.generic_platform import *
+from migen.build.xilinx import XilinxPlatform
+
+
+_io = [
+    ("user_led", 0, Pins("AF19"), IOStandard("LVCMOS25")),
+    ("user_led", 1, Pins("AF23"), IOStandard("LVCMOS25")),
+
+    ("serial", 0,
+        Subsignal("tx", Pins("Y18")),
+        Subsignal("rx", Pins("AA18")),
+        IOStandard("LVCMOS25"),
+    ),
+
+    ("user_sma_clock", 0,
+        Subsignal("p", Pins("AD20"), IOStandard("LVCMOS25"),
+            Misc("DIFF_TERM=TRUE")),
+        Subsignal("n", Pins("AD21"), IOStandard("LVCMOS25"),
+            Misc("DIFF_TERM=TRUE"))
+    ),
+
+    ("clk_gtp", 0,
+        Subsignal("p", Pins("R6")),
+        Subsignal("n", Pins("R5")),
+    ),
+
+    ("clk125_gtp", 0,
+        Subsignal("p", Pins("U6")),
+        Subsignal("n", Pins("U5")),
+    ),
+
+    ("sfp", 0,
+        Subsignal("txp", Pins("AA2")),
+        Subsignal("txn", Pins("AA1")),
+        Subsignal("rxp", Pins("AB4")),
+        Subsignal("rxn", Pins("AB3")),
+    ),
+
+    ("sfp", 1,
+        Subsignal("txp", Pins("W2")),
+        Subsignal("txn", Pins("W1")),
+        Subsignal("rxp", Pins("Y4")),
+        Subsignal("rxn", Pins("Y3")),
+    ),
+
+    ("sfp", 2,
+        Subsignal("txp", Pins("U2")),
+        Subsignal("txn", Pins("U1")),
+        Subsignal("rxp", Pins("V4")),
+        Subsignal("rxn", Pins("V3")),
+    ),
+
+    ("sfp", 3,
+        Subsignal("txp", Pins("R2")),
+        Subsignal("txn", Pins("R1")),
+        Subsignal("rxp", Pins("T4")),
+        Subsignal("rxn", Pins("T3")),
+    ),
+
+    ("clkrec", 0,
+        Subsignal("p", Pins("G7")),
+        Subsignal("n", Pins("F7")),
+        IOStandard("LVDS_18"),
+    ),
+    ("clk_fpgaio", 0,
```

Comment:
Please use signal names consistent with Kasli 2 (which has WRPLL, and Kasli 1 has not).